crates/codegen/parser/src/chumsky/character_filter.rs

```diff
@@ -94,13 +92,20 @@ impl<'context> CharacterFilter<'context> {
                 }
             }
 
-            EBNF::DelimitedBy(_)
-            | EBNF::OneOrMore(_)
-            | EBNF::Optional(_)
-            | EBNF::Repeat(_)
-            | EBNF::SeparatedBy(_)
-            | EBNF::Sequence(_)
-            | EBNF::ZeroOrMore(_) => None,
+            // TODO: only do this if the referenced production is marked as inline
+            EBNF::Reference(name) => {
```

Comment:
Not sure I understand that. Can you please explain how/when it is marked as inline?